### PR TITLE
Re-add nodes with hyphens in nav tree

### LINKED ISSUE #29: Paths with hyphens (-) no longer show up in tree

### PR DESCRIPTION
**Pull request by ricardoboss**
Fixes #29

### CODE REVIEW

File: TRENZ.Docs.API.Test/NavNodeFlaggingServiceTest.cs [hunk 1]
```diff
@@ -34,8 +34,8 @@ public static IEnumerable<object[]> UpdateHasContentFlagValuesProvider()
             },
             new List<ISourceFile>
             {
-                new MemorySourceFile("uid", "root", "root.md", "root content", "root"),
-                new MemorySourceFile("uid2", "nested", Path.Combine("root", "nested.md"), "nested content", "root/nested"),
+                new MemorySourceFile( "root.md", "root content"),
```
Comment (chucker) [inline TRENZ.Docs.API.Test/NavNodeFlaggingServiceTest.cs]: stray space

File: TRENZ.Docs.API/Services/MarkdownFileProcessingService.cs [hunk 2]
```diff
@@ -13,13 +14,15 @@ namespace TRENZ.Docs.API.Services;
 
 public class MarkdownFileProcessingService : IFileProcessingService
 {
+    private static string ToMd5(string text) => Convert.ToHexString(MD5.HashData(Encoding.UTF8.GetBytes(text)));
```
Comment (chucker) [inline TRENZ.Docs.API/Services/MarkdownFileProcessingService.cs:18]: Consider xxHash.

**Review by chucker — CHANGES_REQUESTED** [review]

**Comment by chucker** [comment]
I'm unsure what `Uid` and the MD5 hash _do_. Are they how items get referenced in the index?

**Comment by ricardoboss** [comment]
> I'm unsure what `Uid` and the MD5 hash _do_. Are they how items get referenced in the index?

Meilisearch requires a unique field in the indexed documents. AFAIK, they are not relevant for anything else. Docs: https://docs.meilisearch.com/learn/core_concepts/primary_key.html

**Comment by chucker** [comment]
> > I'm unsure what `Uid` and the MD5 hash _do_. Are they how items get referenced in the index?
> 
> Meilisearch requires a unique field in the indexed documents. AFAIK, they are not relevant for anything else. Docs: https://docs.meilisearch.com/learn/core_concepts/primary_key.html

OK, but

* can't the _path_ be the pkey? wouldn't it have to be unique?
* if we use a hash as the pkey, are there performance (garbage collection?) considerations? If a document gets updated, its hash invariably changes, which means Meilisearch will treat it as a completely _different_ document rather than an update. Does this cause fragmentation issues?

**Comment by ricardoboss** [comment]
The path cannot be the key, because `/` is an invalid character for the key. The idea behind using a hash was to only have valid characters which are still unique for the nodes location.

As for the fragmentation issue: currently, we delete the whole index before re-indexing all files anyway. The location of a file (the only thing influencing the hash) also doesn't change if the content changes.

**Comment by chucker** [comment]
> currently, we delete the whole index before re-indexing all files anyway

Right.

File: TRENZ.Docs.API/Models/NavTree.cs [hunk 3]
```diff
@@ -88,7 +93,7 @@ bool HasPermissionFor(NavNode node, params string[] permissions)
             {
                 var include = includeNode(kvp.Value);
                 if (!include)
-                    onNodeExcluded?.Invoke();
+                    onNodeExcluded?.Invoke(kvp.Value);
```
Comment (chucker) [inline TRENZ.Docs.API/Models/NavTree.cs:97]: kvp.Value is used a lot… refactor?

**Review by chucker — CHANGES_REQUESTED** [review]

File: TRENZ.Docs.API/Services/MarkdownFileProcessingService.cs (hunk 2 above)
Comment (ricardoboss) [reply TRENZ.Docs.API/Services/MarkdownFileProcessingService.cs:18]: Would it be that beneficial to speed up this method? I'd say leave it as it is

**Comment by ricardoboss** [comment]
@chucker we could think about a better way to update the index instead of re-creating it entirely. Then the UID would have more meaning to it. Anyway, neither of those has to do with this PR. I merely refactored them to simplify the source file interface.

**Review by chucker — APPROVED** [review]

**Pull request merged**